webapp/src/shared/components/Input/EditUserForm.tsx

```diff
@@ -2,39 +2,85 @@ import { useState } from 'react';
 import { Button, ButtonVariant, Input, InputVariant } from '../';
 import './EditUserForm.css';
 
-export default function EditUserForm() {
-  const [fullName, setFullName] = useState('');
-  const [email, setEmail] = useState('');
+type EditUserProps = {
+  origFullName: string;
+  origEmail: string;
+};
+
+export default function EditUserForm({
+  origFullName,
+  origEmail,
+}: EditUserProps) {
+  const [fullName, setFullName] = useState(origFullName);
+  const [email, setEmail] = useState(origEmail);
+
+  async function updateData() {
+    const requestOptions = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        fullName,
+        email,
+      }),
+    };
+    await fetch('http://localhost:3000/api/v1/users', requestOptions);
```

Comment:
suggestion (non-blocking): This might be out of the scope of this PR.

We could use the UsersApi service from the generated OpenAPI spec.
We could have a status message which shows after the user submits the form (or a Notification component?).

Something like this:

<details>

```diff
diff --git a/webapp/src/shared/components/Input/EditUserForm.tsx b/webapp/src/shared/components/Input/EditUserForm.tsx
index 6dbe974..5b6b0b9 100644
--- a/webapp/src/shared/components/Input/EditUserForm.tsx
+++ b/webapp/src/shared/components/Input/EditUserForm.tsx
@@ -1,49 +1,61 @@
-import { useState } from 'react';
-import { Button, ButtonVariant, Input, InputVariant } from '../';
+import { useEffect, useState } from 'react';
+import {
+  Button,
+  ButtonVariant,
+  Input,
+  InputVariant,
+  Text,
+  TextColor,
+  TextVariant,
+} from '../';
+import { Configuration, UsersApi } from '../../generated';
 import './EditUserForm.css';
 
+// We could move this part to another file and import it from there. shared/services/ApiService.ts?
+const config = new Configuration({ basePath: window.location.origin });
+const usersApi = new UsersApi(config);
+
 type EditUserProps = {
   origFullName: string;
   origEmail: string;
 };
 
+type EditUserFormStatus = {
+  type: 'success' | 'error';
+  message: string;
+};
+
 export default function EditUserForm({
   origFullName,
   origEmail,
 }: EditUserProps) {
   const [fullName, setFullName] = useState(origFullName);
   const [email, setEmail] = useState(origEmail);
+  const [status, setStatus] = useState<EditUserFormStatus | null>(null);
 
   async function updateData() {
-    const requestOptions = {
-      method: 'PATCH',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        fullName,
-        email,
-      }),
-    };
-    await fetch('http://localhost:3000/api/v1/users', requestOptions);
+    try {
+      await usersApi.userControllerUpdateCurrentUser({
+        updateUserDto: { fullName, email },
+      });
+      setStatus({ type: 'success', message: 'Update successfully' });
+    } catch {
+      setStatus({ type: 'error', message: 'Invalid name or email' });
+    }
   }
 
   const handleOnSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
     e.preventDefault();
-    let txt;
-    if (
-      window.confirm(
-        `Your name: "${fullName}" and your email: "${email}".\n
-        Is this correct?`,
-      )
-    ) {
-      txt = 'Your data has been saved';
-      updateData();
-    } else {
-      txt = 'Canceled!';
-    }
-    alert(txt);
+    updateData();
   };
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setStatus(null);
+    }, 3000);
+    return () => clearTimeout(timer);
+  }, [status]);
+
   return (
     <>
       <form
@@ -73,6 +85,16 @@ export default function EditUserForm({
             }}
           />
         </div>
+        {status && (
+          <Text
+            variant={TextVariant.PARAGRAPH}
+            color={
+              status.type === 'success' ? TextColor.ONLINE : TextColor.OFFLINE
+            }
+          >
+            {status.message}
+          </Text>
+        )}
       </form>
       <div className="edit-user-form-button">
         <Button

```

</detais>

Reply:
If you don't mind I am going to create another issuer and I'll try it